File: src/main/java/com/intellij/xtextLanguage/xtext/generator/EnumRule.java
package com.intellij.xtextLanguage.xtext.generator;

import com.intellij.psi.util.PsiTreeUtil;
import com.intellij.xtextLanguage.xtext.psi.XtextAnnotation;
import com.intellij.xtextLanguage.xtext.psi.XtextEnumRule;

public class EnumRule {
    public XtextEnumRule myRule;
    public String name;
    public String returnType;
    public String alternatives;
    public String[] anotations;

    public EnumRule(XtextEnumRule rule) {
        myRule = rule;
        name = rule.getName();
        returnType = rule.getReturns() != null ? rule.getReturns().getNextSibling().getText() : null;
        alternatives = getAlternativesAsString(myRule);
        XtextAnnotation[] xtextAnnotations = PsiTreeUtil.getChildrenOfType(myRule, XtextAnnotation.class);
        if (xtextAnnotations != null && xtextAnnotations.length > 0) {
            anotations = new String[xtextAnnotations.length];
            int i = 0;
            for (XtextAnnotation xtextAnnotation : xtextAnnotations) {
                anotations[i] = xtextAnnotation.getText();
                i++;
            }
        }
    }

    protected String getAlternativesAsString(XtextEnumRule myRule) {
        String result = "";

        return result;
    }
}

File: src/main/java/com/intellij/xtextLanguage/xtext/generator/models/elements/EnumRule.kt
package com.intellij.xtextLanguage.xtext.generator.models.elements

import com.intellij.psi.util.PsiTreeUtil
import com.intellij.xtextLanguage.xtext.psi.XtextEnumLiteralDeclaration
import com.intellij.xtextLanguage.xtext.psi.XtextEnumRule

class EnumRule(val myRule: XtextEnumRule) : ModelRule() {
    override val name = myRule.validID.text.replace("^", "Caret").capitalize()
    override var returnTypeText: String = myRule.typeRef?.text ?: name
    override val alternativeElements: MutableList<RuleElement> = PsiTreeUtil.findChildrenOfType(myRule, XtextEnumLiteralDeclaration::class.java).map { EnumRuleElement(it) }.toMutableList()
    override var isDataTypeRule = false

}